cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -474,8 +476,11 @@ public void programRunRecord(HttpRequest request, HttpResponder responder,
     }
 
     try {
-      RunRecord runRecord = store.getRun(Id.Program.from(namespaceId, appId, type, programId), runid);
-      if (runRecord != null) {
+      RunRecordMeta runRecordMeta = store.getRun(Id.Program.from(namespaceId, appId, type, programId), runid);
+      if (runRecordMeta != null) {
+        RunRecord runRecord = new RunRecord(runRecordMeta.getPid(), runRecordMeta.getStartTs(),
```

Comment:
Since `RunRecordMeta` extends from `RunRecord`, it's better to have a copy constructor in `RunRecord`. With then you can simple do:

```java
responder.sendJson(HttpResponseStatus.OK, new RunRecord(runRecordMeta));
```